store-implementation/simple-federated-store/src/main/java/uk/gov/gchq/gaffer/federated/simple/FederatedStore.java

```diff
@@ -186,6 +194,59 @@ public void setDefaultGraphIds(final List<String> defaultGraphIds) {
         this.defaultGraphIds = defaultGraphIds;
     }
 
+    /**
+     * Change the graph's ID for the specified graph.
+     *
+     * @param graphToUpdateId the graph that is to have its ID updated
+     * @param newGraphId the new graph ID
+     * @throws OperationException if accumulo table fails to be renamed
+     *
+     */
+    public void changeGraphId(final String graphToUpdateId, final String newGraphId) throws OperationException {
+        try {
+            final GraphSerialisable graphToUpdate = getGraph(graphToUpdateId);
+            updateTableAndCache(graphToUpdate, newGraphId);
+        } catch (final CacheOperationException e) {
+            // Unknown issue getting graph from cache
+            throw new GafferRuntimeException(e.getMessage(), e);
+        }
+    }
+
+    /**
+     * Remove graph from cache, update any Accumulo table names
+     * and then re-add graph with new id back into the cache.
+     *
+     * @param graphToUpdate The graph to have its ID updated
+     * @param newGraphId the new graph ID
+     * @throws OperationException if accumulo table fails to be renamed
+     *
+     */
+    private void updateTableAndCache(final GraphSerialisable graphToUpdate, final String newGraphId) throws OperationException {
+        final String graphToUpdateId = graphToUpdate.getGraphId();
+
+        // Remove from cache
+        removeGraph(graphToUpdateId);
+
+        // Update accumulo tables with new graph ID
+        if (graphToUpdate.getStoreProperties().getStoreClass().startsWith(AccumuloStore.class.getPackage().getName())) {
+            try {
+                renameTable((AccumuloProperties) graphToUpdate.getStoreProperties(), graphToUpdateId, newGraphId);
+            } catch (final StoreException e) {
+                throw new OperationException("Error trying to rename Accumulo table for graph: " + graphToUpdateId, e);
+            }
+        }
+
+        GraphSerialisable updatedGraphSerialisable = new GraphSerialisable.Builder(graphToUpdate)
```

Comment:
Can't you just do somthing like `graphToUpdate.getConfig().setGraphId(newGraphId)` then add back to the cache?